src/verifiers/verificationManager.ts

```diff
@@ -0,0 +1,22 @@
+import { SignedDocument } from "@govtechsg/open-attestation";
+import { VerificationManagerOptions, VerificationFragment, Verifier } from "../types/core";
+
+/**
+ * A verification manager will run a list of {@link Verifier} over a signed document.
+ * Before running each verifier, the manager will make sure the verifier can handle the specific document by calling its exposed test function.
+ * The manager will return the consolidated list of {@link VerificationFragment}
+ */
+export const verificationManager = (verifiers: Verifier[]) => (
+  document: SignedDocument,
+  options: VerificationManagerOptions
+): Promise<VerificationFragment[]> => {
+  const promises: Promise<VerificationFragment>[] = verifiers.map(verifier => {
+    if (verifier.test(document, options)) {
+      return verifier.verify(document, options);
+    }
+    return verifier.skip(document, options);
+  });
+
+  options.promisesCallback?.(promises);
```

Comment:
test it MF

Reply:
will skip test for this one, I'm a bit lazy sorry =). Blame me if there is a bug on this one, I will fix and write tests